iulianIachimov/src/main/java/datacollectiontask/Main.java

```diff
@@ -0,0 +1,123 @@
+        User userTwo = new User("Adam Smith", addressMapTwo);
+
+        // create first list of items
+        List<Item> ItemListOne = new ArrayList<>();
+        Item itemOne = new Item("Iphone", Type.GADGET);  // create itemOne
+        Item itemTwo = new Item("Trousers", Type.CLOTHES); // create itemTwo
+        ItemListOne.add(itemOne); // add itemOne to ItemListOne
+        ItemListOne.add(itemTwo);
+
+        //Create second list of items
+        List<Item> ItemListTwo = new ArrayList<>();
+        Item itemThree = new Item("Trotinette", Type.TOY);  // create itemOne
+        Item itemFour = new Item("Frying Pan", Type.KITCHENWARE); // create itemTwo
+        ItemListTwo.add(itemThree); // add itemOne to ItemListOne
+        ItemListTwo.add(itemFour);
+
+        //Create 2 objects of Order type ()
+        Order orderOne = new Order(userOne, ItemListOne);
+        Order orderTwo = new Order(userTwo, ItemListTwo);
+
+        //Create list of orders
+        List<Order> orderList = new ArrayList<>();
+        orderList.add(orderOne);
+        orderList.add(orderTwo);
+
+        //Filter Orders by Type CLOTHES, calling funcion filterOrderByItemType(orderList)
+        List<Order> filteredOrderList = filterOrderByItemType(orderList);
+        System.out.println("the size of filtered order list is " + filteredOrderList.size());
+
+        /*
+        //collection sort, out of tasks
+        System.out.println("userList before sorting");
+        for (User user: userList){
+            System.out.println(user.username);
+        }
+
+        Collections.sort(userList, new Comparator<User>() {
+            @Override
+            public int compare(User userOne, User userTwo) {
+                return userOne.username.compareTo(userTwo.username);
+            }
+        });
+
+        System.out.println("userList after sorting");
+        for (User user: userList){
+            System.out.println(user.username);
+        }
+    */
+
+
+        // Creating the list for users
+        List<User> userList = new ArrayList<>();
+        userList.add(userOne);
+        userList.add(userTwo);
+
+        // Filter users by city
+        List<User> filteredListofUsers = new ArrayList<>();
+        filteredListofUsers = filterUsersByCity(userList, "Chisinau"); // call for viewing the users from Chisinau city
+        System.out.println("The filtered list of users is " + filteredListofUsers.size());
+    }
+
+    // creating function to filter the orders by type CLOTHES
+    public static List<Order> filterOrderByItemType(List<Order> orderList) {
+        List<Order> orderResult = new ArrayList<>(); // creating a list to store the result
+        for (Order order : orderList) {
+            for (Item item : order.items) {
+                if (Type.CLOTHES == item.type) {
+                    orderResult.add(order);
+                    System.out.println("Order " + order.id + " from " + order.date + " contains type Clothes");
+                    //break;
```

Comment:
Uncomment break statement, otherwise you can get the same order for twice times.